x-pack/docs/en/security/fips-140-compliance.asciidoc

```diff
@@ -87,25 +112,39 @@ keys must have corresponding length according to the following table:
 [discrete]
 ==== Password Hashing
 
-{es} offers a number of algorithms for securely hashing credentials in memory and
-on disk. However, only the `PBKDF2` family of algorithms is compliant with FIPS
-140-2 for password hashing. You must set the `cache.hash_algo` realm settings
+While {es} offers a number of algorithms for securely hashing credentials in memory and
+on disk, only the `PBKDF2` based family of algorithms is compliant with FIPS
+140-2 for password hashing. However, since `PBKDF2` is essentially a key derivation
+function, your JVM security provider may enforce a 112 bit key strength requirement
+(see <<keystore-fips-password>>), and this requirement may affect password hashing
+in {es}, even though FIPS 140-2 does not mandate password standards for user
+authentication. In order to accommodate this requirement, 
+while allowing you to use passwords that satisfy your security policy, {es} offers
+<<hashing-settings, pbkdf2_stretch>> which is the suggested hashing algorithm when running
+{es} in FIPS 140-2 environments. `pbkdf2_stretch` performs a single round of SHA-512
+on the user password before passing it to the `PBKDF2` implementation.
+
+NOTE: You can still use one of the plain `pbkdf2` options instead of `pbkdf2_stretch` if
+you have external policies and tools that can ensure all user passwords for the reserved,
+native and file realms are longer than 14 bytes.
+
+You must set the `cache.hash_algo` realm settings
 and the `xpack.security.authc.password_hashing.algorithm` setting to one of the
-available `PBKDF2` values.
+available `pbkdf_stretch_*` values.
 See <<hashing-settings>>.
 
 Password hashing configuration changes are not retroactive so the stored hashed
-credentials of existing users of the file and native realms will not be updated
-on disk.
-Authentication will still work, but in order to ensure FIPS 140-2 compliance,
-you would need to recreate users or change their password using the
-<<users-command, elasticsearch-user>> CLI tool for the file realm and the
-<<security-api-put-user,create users>> and <<security-api-change-password,change
-password>> APIs for the native realm.
+credentials of existing users of the reserved, native and file realms will not be
```

Comment:
Style nits: Avoid future tense. Use serial commas.

We could probably reword to avoid passive voice too, but that may not be needed.
```suggestion
credentials of existing users of the reserved, native, and file realms are not
```